viewer-prototype/src/browser/trace-viewer/components/table-output-component.tsx

```diff
@@ -2,78 +2,178 @@
+    private columnArray = new Array<any>();
+    private showIndexColumn = false;
+    private components: any;
 
-    constructor(props: AbstractOutputProps) {
+    static defaultProps: Partial<TableOutputProps> = {
+        cacheBlockSize: 200,
+        maxBlocksInCache: 5,
+        columnWidth: 200,
+        blockLoadDebounce: 250,
+        tableHeight: '300px',
+        tableWidth: '100%'
+    };
+
+    constructor(props: TableOutputProps) {
         super(props);
+
+        this.components = {
+            loadingRenderer: (params: any) => {
+                if (params.value !== undefined) {
+                    return params.value;
+                } else {
+                    return '<i class="fa fa-spinner fa-spin"></i>';
+                }
+            }
+        };
+
         this.state = {
             outputStatus: 'RUNNING',
-            tableColumns: [],
-            tableLines: []
+            tableColumns: []
         };
-        this.updateTable();
     }
 
     renderMainArea(): React.ReactNode {
-        return <div id='events-table' className='ag-theme-balham-dark' style={{ height: '300px', width: '100%' }}>
+        return <div id='events-table' className='ag-theme-balham-dark' style={{height: this.props.tableHeight, width: this.props.tableWidth}}>
             <AgGridReact
-                columnDefs={this.state.tableColumns}
-                rowData={this.state.tableLines}>
+                columnDefs={this.columnArray}
+                rowModelType='infinite'
+                cacheBlockSize={this.props.cacheBlockSize}
+                maxBlocksInCache={this.props.maxBlocksInCache}
+                blockLoadDebounceMillis={this.props.blockLoadDebounce}
+                debug={this.debugMode}
+                onGridReady={this.onGridReady}
+                components={this.components}
+            >
             </AgGridReact>
         </div>;
     }
 
-    private async updateTable() {
+    private async fetchTableLines(fetchIndex: number, linesToFetch: number) {
+        const traceUUID = this.props.traceId;
+        const tspClient = this.props.tspClient;
+        const outPutId = this.props.outputDescriptor.id;
+
+        const lineResponse = (await tspClient.fetchTableLines(traceUUID, outPutId, QueryHelper.tableQuery(this.columnIds, fetchIndex, linesToFetch))).getModel();
+        const model = lineResponse.model;
+        const lines = model.lines;
+        const linesArray = new Array<any>();
+        lines.forEach(line => {
+            const obj: any = {};
+            const cells = line.cells;
+            const ids = model.columnIds;
+
+            if (this.showIndexColumn) {
+                obj[0] = line.index.toString();
+            }
+
+            for (let i = 0; i < cells.length; i++) {
+                const id = this.showIndexColumn ? ids[i] + 1 : ids[i];
+                obj[id] = cells[i].content;
+            }
+            linesArray.push(obj);
+        });
+
+        this.setState({
```

Comment:
Is it me or there's not really a state change here? this.columnArray is always the same, and status is not even defined here! (and there's no warning, that worries me...)

Reply:
Well caught! I removed the setState from here. 

outputStatus is a property of AbstractOutputState. It's used in time-graph-component.tsx.